casper/src/test/scala/coop/rchain/casper/util/rholang/RuntimeManagerTest.scala

```diff
@@ -32,18 +38,205 @@ class RuntimeManagerTest extends FlatSpec with Matchers {
     val deploy     = ConstructDeploy.sourceDeployNow(badRholang)
     val (_, Seq(result)) =
       runtimeManager
-        .use(mgr => mgr.computeState(mgr.emptyStateHash, deploy :: Nil))
+        .use(mgr => mgr.computeState(mgr.emptyStateHash)(deploy :: Nil))
         .runSyncUnsafe(10.seconds)
 
     result.status.isFailed should be(true)
   }
 
+  "computeState" should "charge for evaluating deployments" in {
+
+    val (_, genesisPk)   = Ed25519.newKeyPair
+    val (userSk, userPk) = Ed25519.newKeyPair
+
+    // TODO: Ensure that phloPrice and phloLimit are non-negative during deploy validation.
+    // TODO: Extract out genesisTerms into method.
+    // TODO: Compensate for the case that phloPrice max is Long.MaxValue.
+    // TODO: Don't recompute genesisTerms effect for each test.
+    val phloPriceGen: Gen[Long] = Gen.choose(0L, 100L)
+    val phloLimitGen: Gen[Long] = Gen.choose(0L, accounting.MAX_VALUE.toLong)
+    val balanceGen: Gen[Long]   = Gen.choose(0L, Long.MaxValue)
+
+    def genesisTerms(balance: Long): Seq[DeployData] =
+      Seq(
+        StandardDeploys.listOps,
+        StandardDeploys.either,
+        StandardDeploys.nonNegativeNumber,
+        StandardDeploys.makeMint,
+        StandardDeploys.lockbox,
+        StandardDeploys.PoS,
+        StandardDeploys.authKey,
+        StandardDeploys.revVault,
+        StandardDeploys
+          .revGenerator(
+            genesisPk,
+            Seq(Vault(RevAddress.fromPublicKey(userPk).get, initialBalance = balance)),
+            supply = Long.MaxValue
+          ),
+        StandardDeploys
+          .poSGenerator(
+            genesisPk,
+            ProofOfStake(
+              minimumBond = 0L,
+              maximumBond = Long.MaxValue,
+              Seq(Validator(genesisPk, stake = 0L))
+            )
+          )
+      )
+
+    val posId = ByteString.copyFrom(
+      Base16.unsafeDecode("cc87bf7747a8c176714b417ca14a63897b07572876c5e38a7896b6007738ef81")
+    )
+    val program = """ @0!(2) """
+
+    forAll(phloPriceGen, phloLimitGen, balanceGen, minSuccessful(15)) {
+      (phloPrice: Long, phloLimit: Long, balance: Long) =>
+        val deploy0 =
+          SignDeployment.sign(
+            userSk,
+            ConstructDeploy
+              .sourceDeployNow(program)
+              .withDeployer(ByteString.copyFrom(userPk.bytes))
+              .withPhloPrice(phloPrice)
+              .withPhloLimit(phloLimit)
+          )
+        forAll(minSuccessful(1)) { withPayment: Boolean =>
```

Comment:
instead of a generative test over a Boolean domain, with an if/match inside - let's have two tests for two cases. Especially that those should be two different methods for with and without payment.